codespell_lib/_codespell.py

```diff
@@ -493,6 +489,7 @@ def parse_options(
         "accepts globs as well. E.g.: if you want "
         "codespell to skip .eps and .txt files, "
         'you\'d give "*.eps,*.txt" to this option.',
+        default=list(),
```

Comment:
Default arguments shouldn't be mutable.

Reply:
Default arguments of functions sure. But for a `parser` instance that gets created and used once here I don't see a problem with it (it will get mutated as things get appended but that's fine)?

Reply:
Or pass a tuple instead, which is good enough here since we just need to iterate, so anything `Iterable` is OK.

Reply:
You're right, t's not a problem if the function/method is used only once. But linters may raise issue (I think there's a ruff/refurb rule for that). Any way, the fix is so simple that I went ahead and fixed it.

Reply:
The fix is not so simple :grin:

Note that parser.add_argument("-S", ...) might be called multiple times. Isn't that a **potential** problem? I know it is not a problem here, but it does require some knowledge of the inner workings of [ArgumentParser.add_argument](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument), doesn't it?

Reply:
> Note that parser.add_argument("-S", ...) might be called multiple times. 

To be clear I don't think the `add_argument` will be called multiple times but rather that the user might pass multiple `-S`'s, right? Internally `parser` will just call `var.append(...)` which seems fine. If you didn't supply an argument the default is `None` so it would just on the first `-S` create its own list then mutate that.

Think about it this way if we did:
```
mylist = []
parser.add_arg('--whatever', default=mylist)
parser.parse()
```
now you can access the args via `parser.whatever` but that will match `mylist`, i.e. `mylist == parser.whatever` but this isn't a problem (and it's also not a problem that we don't bother to store `mylist` in the local variables and just use `parser.whatever`).

I'm not sure what the fear is here. Typically the "no mutable default" is used for *function arguments* because people didn't realize that doing:
```
def fun(a=list()):
    a.append('b')
    return a
```
would do
```
>>> fun()
['b']
>>> fun()
['b', 'b']
```
because of the mutable default argument of the function, i.e., multiple calls have an unexpected behavior. But we aren't dealing with a mutable function argument here, but rather passing a mutable variable to a method that we know later will have `.append` called on it when we call `parser.parse()` (and we call that once!).

Reply:
You're right of course. This is function **call**, not a function **definition**. The name `default` somehow made me switch to "function definition" mode.

Any way, I think the final code is equivalent in terms of complexity.